src/data/markdown/docs/01 guides/02 Using k6/14 Executors.md

```diff
@@ -0,0 +1,295 @@
+|----------------|--------|----------------------------------------------------------------------------------|-------------|
+| `startTime`    | string | Time offset since test start this scenario should begin execution.               | `"0s"`      |
+| `gracefulStop` | string | Time to wait for iterations to finish executing before stopping them forcefully. | `"30s"`     |
+| `exec`         | string | Name of exported JS function to execute.                                         | `"default"` |
+| `env`          | object | Environment variables specific to this scenario.                                 | `{}`        |
+| `tags`         | object | [Tags](/using-k6/tags-and-groups) specific to this scenario.                     | `{}`        |
+
+
+## Examples
+
+- Execute a constant 200 RPS for 1 minute, allowing k6 to dynamically schedule up to 100 VUs.
+
+Note that in order to reliably achieve a fixed request rate, it's recommended to keep
+the function being executed very simple, with preferably only a single request call,
+and no additional processing or `sleep()` calls.
+
+<div class="code-group" data-props='{"labels": [ "constant-arr-rate.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    constant_arr_rate: {  // arbitrary scenario name
+      executor: 'constant-arrival-rate',
+      rate: 200,
+      timeUnit: '1s',
+      duration: '1m',
+      preAllocatedVUs: 50,
+      maxVUs: 100,
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/');
+}
+```
+
+</div>
+
+
+- Execute a variable RPS test, starting at 50, ramping up to 200 and then back to 0,
+  over a 1 minute period:
+
+<div class="code-group" data-props='{"labels": [ "ramping-arr-rate.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    ramping_arrival: {  // arbitrary scenario name
+      executor: 'ramping-arrival-rate',
+      startRate: 50,
+      timeUnit: '1s',
+      preAllocatedVUs: 50,
+      maxVUs: 100,
+      stages: [
+        { target: 200, duration: '30s' },
+        { target: 0, duration: '30s' },
+      ],
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/');
+}
+```
+
+</div>
+
+
+- Run as many iterations as possible with 50 VUs for 30s, and then run 100 iterations
+  per VU of another scenario.
+
+Note the use of `startTime`, and different `exec` functions for each scenario.
+
+<div class="code-group" data-props='{"labels": [ "multiple-scenarios.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    const_vus: {
+      executor: 'constant-vus',
+      exec: 'scenario1',
+      vus: 50,
+      duration: '30s',
```

Comment:
I would rather not introduce `gracefulStop` here since this example already showcases a couple of features, and `gracefulStop` would need a more thorough explanation. And it wasn't meant to stop at _exactly_ 30s. :)

So let's use `gracefulStop` in a separate example in that new gracefulStop/RampDown section you mentioned.

Reply:
See the `gracefulStop` section added in 72d17e4.